### ISSUE TITLE
softmax problem in contextDesc

### ISSUE THREAD

**Issue by yvaine** [lzx551402/tfmatch#9]
Hi,
In contextDesc paper, I saw 
![Screenshot from 2020-09-01 17-37-20](https://user-images.githubusercontent.com/9897879/91833841-cfd0ff80-ec79-11ea-98ae-0505c771b820.png)
in equation(5), the s is softmax(2-D), but the code in line170 of loss.py, list below
        softmax_row = tf.nn.softmax(log_scale * dist_mat, axis=2)
the dist_mat has not subtract by 2, I don't know there is any wrong or i think wrong?

Thanks
